velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -1277,6 +1277,79 @@ TEST_F(DateTimeFunctionsTest, millisecondTimestampWithTimezone) {
           "millisecond(c0)", -980, "+05:30"));
 }
 
+TEST_F(DateTimeFunctionsTest, lastDayOfMonth) {
+  const auto lastDayOfMonth = [&](std::optional<StringView> timestamp) {
+    auto result = evaluateOnce<Date>(
+        "last_day_of_month(c0)",
+        timestamp.has_value()
+            ? std::optional(util::fromTimestampString(timestamp.value()))
+            : std::nullopt);
+    return result.has_value() ? std::optional<std::string>(result)
+                              : std::nullopt;
+  };
+  EXPECT_EQ(std::nullopt, lastDayOfMonth(std::nullopt));
+  EXPECT_EQ("1970-01-31", lastDayOfMonth("1970-01-01 00:00:00"));
+  EXPECT_EQ("2012-01-31", lastDayOfMonth("2012-01-01 00:00:00"));
+  EXPECT_EQ("2012-01-31", lastDayOfMonth("2012-01-31 12:00:00"));
+
+  setQueryTimeZone("Pacific/Apia");
+
+  EXPECT_EQ(std::nullopt, lastDayOfMonth(std::nullopt));
+  EXPECT_EQ("1969-12-31", lastDayOfMonth("1970-01-01 00:00:00"));
+  EXPECT_EQ("2012-01-31", lastDayOfMonth("2012-01-01 00:00:00"));
+  EXPECT_EQ("2012-02-29", lastDayOfMonth("2012-01-31 12:00:00"));
```

Comment:
I'm puzzled by this test case. Why is last day of month for '2012-01-31 12:00:00' is Feb 29 and not Jan 31?

Reply:
I see Jan 31 in Presto:

```
presto> select last_day_of_month(timestamp '2012-01-31 12:00:00 Pacific/Apia');
   _col0
------------
 2012-01-31
(1 row)
```

Reply:
Wikipedia tells us that Pacific/Apia changes from UTC-11 to UTC+13 in 2012:

https://en.m.wikipedia.org/wiki/Time_in_Samoa

But I don't know why Presto behaves differently.

Reply:
> Wikipedia tells us that Pacific/Apia changes from UTC-11 to UTC+13 in 2012:

Can you explain what that means and why the result about is Feb?

> But I don't know why Presto behaves differently.

We need to make sure to implement Presto semantics unless there is a bug in Presto, which we need to fix.

Reply:
It seems `last_day_of_month` converts `2012-01-31 12:00 UTC` to Pacific/Apia time which is `2012-02-01 02:00:00`

```
presto> SELECT timestamp '2012-01-31 12:00 UTC' AT TIME ZONE 'Pacific/Apia';
                _col0                 
--------------------------------------
 2012-02-01 02:00:00.000 Pacific/Apia 
(1 row)
```

Reply:
@zhenghao-bot Thank you for explaining. I get it. It might be helpful to explain this in the documentation of the function and provide an example. Also, it looks like the results would be different if we call last_day_of_month on timestamp with time zone value. 

```
presto> SELECT last_day_of_month(timestamp '2012-01-31 12:00 UTC' AT TIME ZONE 'Pacific/Apia');
   _col0
------------
 2012-02-29
(1 row)


presto> select last_day_of_month(timestamp '2012-01-31 12:00:00 Pacific/Apia');
   _col0
------------
 2012-01-31
(1 row)
```

It looks like we have a test case for the first query, but not the second query. Would it make sense to add it?